service/src/main/java/greencity/service/EventServiceImpl.java

```diff
@@ -137,14 +127,105 @@ public PageableAdvancedDto<EventDto> getAll(Pageable page, Principal principal)
     }
 
     @Override
-    public PageableAdvancedDto<EventDto> getAllUserEvents(Pageable page, String email) {
+    public PageableAdvancedDto<EventDto> getAllUserEvents(
+        Pageable page, String email, String userLatitude, String userLongitude, String eventType) {
         User attender = modelMapper.map(restClient.findByEmail(email), User.class);
-        Page<Event> events = eventRepo.findAllByAttender(page, attender.getId());
-        PageableAdvancedDto<EventDto> eventDtos = buildPageableAdvancedDto(events);
-        setSubscribes(events, eventDtos, attender);
+        List<Event> events = sortUserEventsByEventType(eventType, attender, userLatitude, userLongitude);
+        Page<Event> eventPage = new PageImpl<>(events, page, events.size());
+        PageableAdvancedDto<EventDto> eventDtos = buildPageableAdvancedDto(eventPage);
+        setSubscribes(eventPage, eventDtos, attender);
         return eventDtos;
     }
 
+    private List<Event> sortUserEventsByEventType(
+        String eventType, User attender, String userLatitude, String userLongitude) {
+        if (eventType.equalsIgnoreCase("ONLINE")) {
+            return getOnlineUserEventsSortedByDate(attender);
+        }
+
+        if (eventType.equalsIgnoreCase("OFFLINE")) {
+            if (!userLatitude.isBlank() && !userLongitude.isBlank()) {
```

Comment:
```suggestion
        if (eventType.equalsIgnoreCase("OFFLINE") && !userLatitude.isBlank() && !userLongitude.isBlank()) {
```

Reply:
I have two cases to take into consideration when it gous about OFFLINE. First, when user share geo location and userLatitude and userLongitude come from frontend, the other,this params are empty string as user douesn't share geo location. So I desided to change this way:                                                                              if (eventType.equalsIgnoreCase("OFFLINE")) {
            return (!userLatitude.isBlank() && !userLongitude.isBlank()) ?
                    getOfflineUserEventsSortedCloserToUserLocation(attender, userLatitude, userLongitude) :
                    getOfflineUserEventsSortedByDate(attender);
        }